README.md

```diff
@@ -5,12 +5,57 @@
 
 `amphp/aerys-session` is a non-blocking session handler for use with [`amphp/aerys`](https://github.com/amphp/aerys).
 
-**Required PHP Version**
+## Requirements
 
-- PHP 7+
+- PHP 7
 
-**Installation**
+## Installation
 
 ```bash
 $ composer require amphp/aerys-session
 ```
+
+## Registering the Session Middleware
+
+```php
+$redis = new Amp\Redis\Client("tcp://localhost:6379");
+$mutex = new Amp\Redis\Mutex("tcp://localhost:6379");
+
+$router->use(Aerys\session([
+    "driver" => new Aerys\Session\Redis($redis, $mutex)
+]));
+```
+
+## Using the Session
+
+```php
+public function respondToRequest(Request $request, Response $response) {
+    $session = yield (new Aerys\Session($request))->read();
+
+    if (!$session->has("user")) {
+        $response
+            ->setStatus(401)
+            ->end("Unauthorised");
+    }
+
+    $user = $session->get("user");
+
+    // ...
+
+    $session->set("token", $token);
```

Comment:
Write access only works after `open()`ing the session. `read()` only opens for reading.